source/darwin/dispatch_queue_event_loop.c

```diff
@@ -0,0 +1,496 @@
+    aws_thread_decrement_unjoined_count();
+}
+
+/** Return a aws_string* with unique dispatch queue id string. The id is In format of
+ * "com.amazonaws.commonruntime.eventloop.<UUID>"*/
+static struct aws_string *s_get_unique_dispatch_queue_id(struct aws_allocator *alloc) {
+    struct aws_uuid uuid;
+    AWS_FATAL_ASSERT(aws_uuid_init(&uuid) == AWS_OP_SUCCESS);
+    char uuid_str[AWS_UUID_STR_LEN] = {0};
+    struct aws_byte_buf uuid_buf = aws_byte_buf_from_array(uuid_str, sizeof(uuid_str));
+    uuid_buf.len = 0;
+    aws_uuid_to_str(&uuid, &uuid_buf);
+    struct aws_byte_cursor uuid_cursor = aws_byte_cursor_from_buf(&uuid_buf);
+
+    struct aws_byte_buf dispatch_queue_id_buf;
+    aws_byte_buf_init_copy_from_cursor(
+        &dispatch_queue_id_buf, alloc, aws_byte_cursor_from_c_str("com.amazonaws.commonruntime.eventloop."));
+
+    aws_byte_buf_append_dynamic(&dispatch_queue_id_buf, &uuid_cursor);
+
+    struct aws_string *result = aws_string_new_from_buf(alloc, &dispatch_queue_id_buf);
+    aws_byte_buf_clean_up(&dispatch_queue_id_buf);
+    return result;
+}
+
+/* Setup a dispatch_queue with a scheduler. */
+struct aws_event_loop *aws_event_loop_new_dispatch_queue_with_options(
+    struct aws_allocator *alloc,
+    const struct aws_event_loop_options *options) {
+    AWS_PRECONDITION(options);
+    AWS_PRECONDITION(options->clock);
+
+    struct aws_event_loop *loop = aws_mem_calloc(alloc, 1, sizeof(struct aws_event_loop));
+
+    AWS_LOGF_DEBUG(AWS_LS_IO_EVENT_LOOP, "id=%p: Initializing dispatch_queue event-loop", (void *)loop);
+    if (aws_event_loop_init_base(loop, alloc, options->clock)) {
+        goto clean_up_loop;
+    }
+
+    struct dispatch_loop *dispatch_loop = aws_mem_calloc(alloc, 1, sizeof(struct dispatch_loop));
+    aws_ref_count_init(&dispatch_loop->ref_count, loop, s_dispatch_event_loop_destroy);
+
+    dispatch_loop->dispatch_queue_id = s_get_unique_dispatch_queue_id(alloc);
+
+    dispatch_loop->dispatch_queue =
+        dispatch_queue_create((char *)dispatch_loop->dispatch_queue_id->bytes, DISPATCH_QUEUE_SERIAL);
+    if (!dispatch_loop->dispatch_queue) {
+        AWS_LOGF_FATAL(AWS_LS_IO_EVENT_LOOP, "id=%p: Failed to create dispatch queue.", (void *)loop);
+        aws_raise_error(AWS_ERROR_SYS_CALL_FAILURE);
+        goto clean_up_dispatch;
+    }
+
+    dispatch_loop->synced_data.scheduling_state.is_executing_iteration = false;
+    dispatch_loop->allocator = alloc;
+
+    int err = aws_task_scheduler_init(&dispatch_loop->scheduler, alloc);
+    if (err) {
+        AWS_LOGF_ERROR(AWS_LS_IO_EVENT_LOOP, "id=%p: Initializing task scheduler failed", (void *)loop);
+        goto clean_up_dispatch;
+    }
+
+    aws_linked_list_init(&dispatch_loop->local_cross_thread_tasks);
+    aws_linked_list_init(&dispatch_loop->synced_data.scheduling_state.scheduled_services);
+    aws_linked_list_init(&dispatch_loop->synced_data.cross_thread_tasks);
+
+    dispatch_loop->wakeup_schedule_needed = true;
```

Comment:
Nice catch. This need to be cleaned up. 
The variable was used to avoid explicitly scheduling a thread iteration. Now we used the function `should_schedule_iteration() ` to determine if we should schedule iterations. This is no long in use.